File: ApexToDotNet.API/Models/StrategicPlannerModels.cs
namespace ApexToDotNet.API.Models
{
    public class StrategicPlannerModels
    {
        // Enums
        public enum Priority
        {
            P1 = 1,
            P2 = 2,
            P3 = 3,
            P4 = 4,
            P5 = 5
        }

        public enum Size
        {
            XS,
            S,
            M,
            L,
            XL
        }

        // Main Models
        public class Project
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public Priority Priority { get; set; }
            public string? Status { get; set; }
            public DateTime? StartDate { get; set; }
            public DateTime? EndDate { get; set; }
            public int? AreaId { get; set; }
            public string? AreaName { get; set; }
            public int? InitiativeId { get; set; }
            public string? InitiativeName { get; set; }
            public int? PersonCount { get; set; }
            public int? ActivityCount { get; set; }
            public DateTime? CreatedDate { get; set; }
            public DateTime? UpdatedDate { get; set; }
        }

        public class Area
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public string? Color { get; set; }
            public int? ProjectCount { get; set; }
            public int? InitiativeCount { get; set; }
        }

        public class Initiative
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public Priority Priority { get; set; }
            public Size? Size { get; set; }
            public int? AreaId { get; set; }
            public string? AreaName { get; set; }
            public DateTime? StartDate { get; set; }
            public DateTime? EndDate { get; set; }
            public int? ProjectCount { get; set; }
            public int? ActivityCount { get; set; }
        }

        public class Activity
        {
            public int Id { get; set; }
            public string Title { get; set; } = string.Empty;
            public string? Description { get; set; }
            public Priority Priority { get; set; }
            public Size? Size { get; set; }
            public string? Status { get; set; }
            public int? ProjectId { get; set; }
            public string? ProjectName { get; set; }
            public int? InitiativeId { get; set; }
            public string? InitiativeName { get; set; }
            public int? AssignedToId { get; set; }
            public string? AssignedToName { get; set; }
            public DateTime? DueDate { get; set; }
            public DateTime? CompletedDate { get; set; }
        }

        public class Person
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Email { get; set; }
            public string? Role { get; set; }
            public string? Department { get; set; }
            public int? ProjectCount { get; set; }
            public int? ActivityCount { get; set; }
            public int? CompletedActivityCount { get; set; }
        }

        public class ProjectGroup
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public int? ProjectCount { get; set; }
        }

        public class PersonGroup
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public int? PersonCount { get; set; }
        }

        public class Release
        {
            public int Id { get; set; }
            public string Name { get; set; } = string.Empty;
            public string? Description { get; set; }
            public DateTime? ReleaseDate { get; set; }
            public string? Status { get; set; }
            public int? ProjectCount { get; set; }
            public int? ActivityCount { get; set; }
        }

        // Response Models
        public class NavigationCounts
        {
            public int Projects { get; set; }
            public int Areas { get; set; }
            public int Initiatives { get; set; }
            public int Activities { get; set; }
            public int People { get; set; }
            public int ProjectGroups { get; set; }
            public int PersonGroups { get; set; }
            public int Releases { get; set; }
        }

        public class SearchResult<T>
        {
            public List<T> Items { get; set; } = new();
            public int TotalCount { get; set; }
            public int Page { get; set; }
            public int PageSize { get; set; }
        }
    }
}

File: ApexToDotNet.Web/src/app/models/strategic-planner.models.ts
// Strategic Planner Models
// Aligned with the .NET API response shapes from StrategicPlannerController.cs

// ─── Projects (SP_PROJECTS) ───────────────────────────────────────
// GET /api/sp/SpProjects returns this shape
export interface Project {
  id: number;
  projectName: string;
  owner: string;
  ownerId?: number;
  area?: string;
  initiative?: string;
  initiativeId?: number;
  release?: string;
  releaseId?: number;
  status?: string;
  statusId?: number;
  priority?: string;       // e.g. "P1", "P2"
  priorityId?: number;
  pctComplete?: number;
  targetComplete?: string; // ISO date
  projectSize?: string;    // XS, S, M, L, XL
  tags?: string;
  updated?: string;        // ISO date
}

// Full project for create/edit (matches SpProject.cs)
export interface ProjectDetail {
  id: number;
  projectName: string;
  description?: string;
  ownerId?: number;
  initiativeId?: number;
  releaseId?: number;
  statusId?: number;
  priorityId?: number;
  pctComplete?: number;
  targetComplete?: string;
  projectSize?: string;
  tags?: string;
  focusAreaId?: number;
  url?: string;
  linkName?: string;
  archivedYn?: string;
  // Navigation properties returned by GET /{id}
  owner?: TeamMember;
  initiative?: Initiative;
  release?: Release;
  status?: ProjectStatus;
  priority?: ProjectPriority;
}

// ─── Initiatives (SP_INITIATIVES) ─────────────────────────────────
export interface Initiative {
  id: number;
  initiativeName: string;
  objective?: string;
  area?: string;
  areaId?: number;
  sponsor?: string;
  sponsorId?: number;
  activeProjects?: number;
  statusScale?: number;
  updated?: string;
  // Full entity fields (from GET /{id})
  archivedYn?: string;
}

// ─── Areas (SP_AREAS) ─────────────────────────────────────────────
export interface Area {
  id: number;
  areaName: string;
  description?: string;
  color?: string;
  initiatives?: number;
  projects?: number;
}

// ─── Releases (SP_RELEASE_TRAINS) ─────────────────────────────────
export interface Release {
  id: number;
  releaseTrain: string;
  releaseName: string;
  releaseOwner?: string;
  releaseTargetDate?: string;
  releaseOpenDate?: string;
  releaseCompleted?: string;
  releaseOpenCompleted?: string;
  releaseType?: string;
  projects?: number;
  avgPctComplete?: number;
  updated?: string;
}

// ─── Team Members (SP_TEAM_MEMBERS) ───────────────────────────────
export interface TeamMember {
  id: number;
  firstName: string;
  lastName: string;
  fullName?: string;
  email?: string;
  isCurrentYn?: string;
  jobTitle?: string;
  department?: string;
}

// ─── Activities (SP_ACTIVITIES) ───────────────────────────────────
export interface Activity {
  id: number;
  comments?: string;
  project?: string;
  projectId?: number;
  person?: string;
  teamMemberId?: number;
  startDate?: string;
  endDate?: string;
  tags?: string;
  updated?: string;
}

// ─── Lookups ──────────────────────────────────────────────────────
export interface ProjectStatus {
  id: number;
  status: string;
  displaySeq?: number;
  includeYn?: string;
}

export interface ProjectPriority {
  id: number;
  priority: number;
  description?: string;
}

export interface ActivityType {
  id: number;
  activityTypeName: string;
  isActiveYn?: string;
  displaySequence?: number;
}

export interface TaskType {
  id: number;
  taskTypeName: string;
}

// ─── Utility types ────────────────────────────────────────────────
export interface NavigationCounts {
  projects: number;
  areas: number;
  initiatives: number;
  activities: number;
  people: number;
  releases: number;
  projectGroups: number;
  peopleGroups: number;
}

// ─── Project Groups (SP_PROJECT_GROUPS) ───────────────────────────
export interface ProjectGroup {
  id: number;
  groupName: string;
  description?: string;
  projects?: number;
  updated?: string;
}

// ─── People Groups (SP_GROUPS + SP_GROUP_MEMBERS) ────────────────
export interface PeopleGroup {
  id: number;
  groupName: string;
  groupDescription?: string;
  groupTag?: string;
  members?: number | PeopleGroupMember[];
  updated?: string;
}

export interface PeopleGroupMember {
  id: number;
  teamMemberId: number;
  name?: string;
  groupLeaderYn?: string;
  addedToGroup?: string;
}